src/app/screens/Register/index.tsx

```diff
@@ -6,37 +7,51 @@ import AuthWrapper from '~components/AuthWrapper';
 import Input from '~components/Input';
 import { emailRegex } from '~utils/inputValidations';
 import { AUTH_FIELDS } from '~constants/fields';
+import { useLazyRequest } from '~app/hooks/useRequest';
+import { apiRegister } from '~services/Auth';
+import AlertMessage from '~components/AlertMessage';
+import { User } from '~utils/types';
 
 import styles from './styles.module.scss';
 
-interface DataForm {
-  email: string;
-  firstName: string;
-  lastName: string;
-  locale: string;
-  password: string;
-  passwordConfirmation: string;
-}
-
 function Register() {
-  const { register, handleSubmit, errors, formState, watch } = useForm<DataForm>({
+  const { register, handleSubmit, errors, formState, watch } = useForm<User>({
     mode: 'all'
   });
 
+  const validatePassword = (passwordConfirmation: string) =>
+    passwordConfirmation === watch('password') ? true : 'FormValidations:passwordNotMatch';
+
+  const [state, loading, error, sendRequest] = useLazyRequest({
+    request: apiRegister
+  });
+
   const onSubmit = handleSubmit(data => {
     data.locale = i18next.language;
-    const user = {
-      user: data
-    };
-    console.log(user);
+    data.first_name = data.firstName;
+    data.last_name = data.lastName;
+    data.password_confirmation = data.passwordConfirmation;
```

Comment:
do not mutate data object, but when you use `cerealizr` you dont need to do this